apitest/scripts/trade-simulation-utils.sh

```diff
@@ -0,0 +1,174 @@
+#! /bin/bash
+
+# This file must be sourced by the main driver.
+
+. "${APITEST_SCRIPTS_HOME}/trade-simulation-env.sh"
+
+printdate() {
+    echo "[$(date)] $@"
+}
+
+printbreak() {
+    echo ""
+    echo ""
+}
+
+printcmd() {
+    echo -en "$@\n"
+}
+
+sleeptraced() {
+    PERIOD=$1
+    printdate "sleeping for $PERIOD"
+    sleep "$PERIOD"
+}
+
+commandalert() {
+    # Used in a script function when it needs to fail early with an error message, & pass the error code to the caller.
+    # usage: commandalert <$?> <msg-prefix>
+    if [ "$1" -ne 0 ]
+    then
+        printdate "Error: $2" >&2
+        exit "$1"
+    fi
+}
+
+# TODO rename exitonalert ?
+exitoncommandalert() {
+    # Used in a parent script when you need it to fail immediately, with no error message.
+    # usage: exitoncommandalert <$?>
+    if [ "$1" -ne 0 ]
+    then
+        exit "$1"
+    fi
+}
+
+registerdisputeagents() {
+    # Silently register dev dispute agents.  It's easy to forget.
+    REG_KEY="6ac43ea1df2a290c1c8391736aa42e4339c5cb4f110ff0257a13b63211977b7a"
+    CMD="${CLI_BASE} --port=${ARBITRATOR_PORT} registerdisputeagent --dispute-agent-type=mediator --registration-key=${REG_KEY}"
+    SILENT=$($CMD)
+    commandalert $? "Could not register dev/test mediator."
+    CMD="${CLI_BASE} --port=${ARBITRATOR_PORT} registerdisputeagent --dispute-agent-type=refundagent --registration-key=${REG_KEY}"
+    SILENT=$($CMD)
+    commandalert $? "Could not register dev/test refundagent."
+    # Do something with $SILENT to keep codacy happy.
+    echo "$SILENT"  > /dev/null
+}
+
+getbtcoreaddress() {
+    CMD="bitcoin-cli -regtest  -rpcport=19443 -rpcuser=apitest -rpcpassword=apitest getnewaddress"
+    NEW_ADDRESS=$(${CMD})
```

Comment:
`${` is inconsistent with some other variable references. It's usually used when you have something like `${var}postifx` (you couldn't do `$varpostfix`). `$($CMD)` should be enough here. Though either is fine, whatever you think is more readable, it's just that it would be nice if the choice were consistent.

Reply:
Fixed in commit cb96f27f1.